File: split-bill-api/src/main/java/com/shashankbhat/splitbill/repository/BillShareRepository.java
package com.shashankbhat.splitbill.repository;

import com.shashankbhat.splitbill.entity.BillShareEntity;
import org.springframework.data.domain.Sort;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface BillShareRepository extends JpaRepository<BillShareEntity, Integer> {
    List<BillShareEntity> findAllByBillId(Integer billId, Sort sort);
}

File: app/src/main/java/com/shashankbhat/splitbill/repository/local/BillShareRepository.kt
package com.shashankbhat.splitbill.repository.local

import com.shashankbhat.splitbill.room_db.dao.BillShareDao
import com.shashankbhat.splitbill.room_db.entity.BillShare

class BillShareRepository(private val billShareDao: BillShareDao) {

    suspend fun insert(billShare: BillShare) {
        billShareDao.insert(billShare)
    }

    suspend fun getBillShareByBillId(billId: Int): List<BillShare> {
        return billShareDao.getBillShareByBillId(billId)
    }

    suspend fun delete(billShare: BillShare) {
        billShareDao.delete(billShare)
    }
}